src/test/packages/25-manifest-adoption/deployment.yaml

```diff
@@ -23,7 +23,7 @@ spec:
         app: game
     spec:
       containers:
-        - name: multi-game
+        - name: "multi-game-long-name-to-test-agent-truncate-annotation-name"
```

Comment:
Went back and forth on whether I should add an e2e test. I decided a new e2e test would be heavy, but the pod unit tests won't enforce the 63 character rule without an actual cluster behind them, so I lengthened the name here, so we have a real test that the functionality is now working.

I made sure this test failed when run on main

Reply:
Sounds good, coverage is valuable here and we can always delete later if we need

Reply:
Moving this to test 36, as the container name needs to be the same here otherwise the deployment won't get adopted correctly